packages/cozy-clisk/src/contentscript/ContentScript.spec.js

```diff
@@ -200,4 +199,90 @@ describe('ContentScript', () => {
       ).rejects.toThrow('Timeout error')
     })
   })
+  describe('shouldFullSync', () => {
+    it('should force full sync => Flag', async () => {
+      const contentScript = new ContentScript()
+      contentScript.contentScriptType = PILOT_TYPE
+
+      const options = {
+        flags: {
+          'clisk.force-full-sync': true
+        },
+        trigger: {
+          current_state: {
+            last_execution: '2024-07-24T14:55:57.83761233+02:00',
+            last_failure: '2024-07-23T14:55:57.83761233+02:00',
+            last_success: '2024-07-24T14:55:57.83761233+02:00'
+          }
+        }
+      }
+      const result = await contentScript.shouldFullSync(options)
+      expect(result).toEqual({
+        forceFullSync: true,
+        distanceInDays: 0
+      })
+    })
+    it('should force full sync => First execution', async () => {
+      const contentScript = new ContentScript()
+      contentScript.contentScriptType = PILOT_TYPE
+
+      const options = {
+        flags: {
+          'clisk.force-full-sync': false
+        },
+        trigger: {
+          current_state: {}
+        }
+      }
+      const result = await contentScript.shouldFullSync(options)
+      expect(result).toEqual({
+        forceFullSync: true,
+        distanceInDays: 0
+      })
+    })
+    it('should force full sync => Last execution failed', async () => {
+      const contentScript = new ContentScript()
+      contentScript.contentScriptType = PILOT_TYPE
+
+      const options = {
+        flags: {
+          'clisk.force-full-sync': false
+        },
+        trigger: {
+          current_state: {
+            last_execution: '2024-07-24T14:55:57.83761233+02:00',
+            last_failure: '2024-07-24T14:55:57.83761233+02:00',
+            last_success: '2024-07-23T14:55:57.83761233+02:00'
+          }
+        }
+      }
+      const result = await contentScript.shouldFullSync(options)
+      expect(result).toEqual({
+        forceFullSync: true,
+        distanceInDays: 0
+      })
+    })
+    it('should not force full sync', async () => {
```

Comment:
Tu peux améliorer cette description et dire quel cas tu couvres ? `should not force full sync when ...`

Reply:
Si tu penses que c'est nécéssaire ok, mais pour m'expliqué je partais du principe que si on est pas dans un des 3 cas au dessus (fullSync par user, fullSync par erreur ou fullSync par distanceInDays) le dernier cas est finalement "le cas nominal" quand tout se passe bien (aka quand l'utilisateur n'as pas explicitement demander une fullSync, quand on a fait le dernier tour en succès ou quand ça fait moins d'un mois depuis la dernière fois). Ca me paraissait assez evident, mais du coup je me demande si il faudrait pas rajouter les 3 positifs à ce moment là ?

Reply:
Tu peux dans ce cas juste ajouter "in nominal case"